### PR TITLE
Screen Names Update and Unique Names Logic

### PR DESCRIPTION
**Pull request by phoebexy**
Updated logic for names

### CODE REVIEW

File: server/controllers/batches.ts [hunk 1]
```diff
@@ -61,12 +57,22 @@ export const joinBatch = async function(data, socket, io) {
...
+          if (
+            totalBatchUsers > randomAnimal.length &&
+            user.nickname === nickname
+          ) {
+            nickname = false;
+          } else if (
+            user.nickname.includes(animal) ||
+            user.nickname.includes(adjective)
+          ) {
             nickname = false;
```
Comment (phoebexy) [inline server/controllers/batches.ts:76]: Logic for setting unique real names

File: server/controllers/batches.ts [hunk 2]
```diff
@@ -656,31 +662,51 @@ export const payStartBonus = async (users, batch) => {
...
+    // don't use animals on any of the users' real nicknames
     team.users.forEach((user) => {
-      let partner = users[user].realNick.slice();
-      for (let i = 0; i < partner.length; i++) {
-        if (partner[i] === partner[i].toUpperCase()) {
-          index = i;
-          break;
-        }
-      }
-      teamAnimals.push(partner.slice(0, index));
```
Comment (phoebexy) [inline server/controllers/batches.ts:668]: Below: Logic for setting unique real names, for every new round, subtract all animals used in user real names and previously seen nick names, if this results in less new names than required in the round, then we use the original full list and just avoid using the same two animals in one round

**Review by phoebexy — COMMENTED** [review]
commented the logic changes

File: README.md [hunk 3]
```diff
@@ -76,20 +77,22 @@ The experiment works by generating several chat rooms, each of `TEAM_SIZE` peopl
    3. post chat activities
 3. post experiment activities
 
-There are usually several rounds of chats and critically, some are with prior teams and some are with random teams. We can can control and randomize when this happens. The user names of each individual are randomized with `makeName` from `tools.ts` and the team configurations are governed by `createTeams` in the same file.
+There are usually several rounds of chats and critically, some are with prior teams and some are with random teams. We can can control and randomize when this happens. The user names of each individual are randomized and the team configurations are governed by `createTeams` in the `utils.ts` file.
```
Comment (markwhiting) [inline README.md:80]: Great!

File: server/controllers/batches.ts (hunk 2 above)
Comment (markwhiting) [reply server/controllers/batches.ts:668]: Sounds good!

**Review by markwhiting — APPROVED** [review]
Looks good. Please pull in any updates before your final commit because I think there was one earlier today, but otherwise it looks like it will work well. Thanks!

**Pull request merged**